module/zfs/dmu_objset.c

```diff
@@ -1393,6 +1393,20 @@ dmu_objset_sync(objset_t *os, zio_t *pio, dmu_tx_t *tx)
 		os->os_freed_dnodes = 0;
 	}
 
+	if (dmu_objset_ds(os)) {
+		if (dmu_objset_ds(os)->ds_dir->dd_inherit_parent) {
+			mutex_enter(&dmu_objset_ds(os)->ds_dir->
+			    dd_inherit_parent->dd_lock);
+			dmu_objset_ds(os)->ds_dir->dd_inherit_parent->
+			    dd_qos_size = 0;
+			dmu_objset_ds(os)->ds_dir->dd_inherit_parent->
+			    dd_qos_ts = gethrtime();
+			mutex_exit(&dmu_objset_ds(os)->ds_dir->
+			    dd_inherit_parent->dd_lock);
```

Comment:
This is clever, although we probably should say that bandwidth is only bound within the transaction group commit. If we were limiting ourselves to 10MB/sec, it looks to me like we can do a single 4KB write at the start of the txg commit, and do a 50MB write at the end of the txg_commit.